### PR TITLE
Minimal enclosing sphere backport to 2021.2

### PR DESCRIPTION
**Pull request by nigeljw-unity**
Backport of minimal enclosing sphere solution for shadow frustum culling.

https://github.com/Unity-Technologies/Graphics/pull/4436

### Checklist for PR maker
- [ ] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [x] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR. If you do add documentation, make sure to add the relevant Graphics Docs team member as a reviewer of the PR. If you are not sure which person to add, see the [Docs team contacts sheet](https://docs.google.com/spreadsheets/d/1rgUWWgwLFEHIQ3Rz-LnK6PAKmbM49DZZ9al4hvnztOo/edit#gid=1058860420).
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR

Fixes bug where shadows disappear in the frustum corners from false positives in the shadow culling.
fogbugz.unity3d.com/f/cases/1153151

---
### Testing status

Tested with URP default template and custom cascade positioning.

---
### Comments to reviewers

Dependent on this change to trunk.
https://ono.unity3d.com/unity/unity/pull-request/131872/_/2021.2/minimal_enclosing_cascade_sphere

### CODE REVIEW

File: com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs [hunk 1]
```diff
@@ -913,6 +918,18 @@ public bool useAdaptivePerformance
             set { m_UseAdaptivePerformance = value; }
         }
 
+        public bool conservativeEnclosingSphere
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs:921]: missing public API docs.

File: com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs [hunk 2]
```diff
@@ -103,6 +103,11 @@ public static float maxRenderScale
             get => 2.0f;
         }
 
+        public static int maxNumIterationsEnclosingSphere
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs:106]: Missing public API docs.

File: com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs [hunk 3]
```diff
@@ -913,6 +918,18 @@ public bool useAdaptivePerformance
...
+        {
+            get { return m_ConservativeEnclosingSphere; }
+            set { m_ConservativeEnclosingSphere = value; }
+        }
+
+        public int numItertionsEnclosingSphere
```
Comment (sandy-carter-unity) [inline com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs:927]: Missing an "a" in "Itertions"

File: com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs [hunk 4]
```diff
@@ -913,6 +918,25 @@ public bool useAdaptivePerformance
             set { m_UseAdaptivePerformance = value; }
         }
 
+        /// <summary>
+        /// Set to true to enable a conservative method for calculating the size and position of the minimal enclosing sphere around the frustum cascade corner points for shadow culling.
+        /// </summary>
```
Comment (nigeljw-unity) [inline com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs:923]: @oleks-k can you review this?

File: com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs (hunk 3 above)
Comment (nigeljw-unity) [reply com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs]: Thank you for catching this!!!

**Review by phi-lira — APPROVED** [review]

**Review by simon-engelbrecht-soerensen — APPROVED** [review]
Ran it through the same tests as for the original version, and toggling on enclosing spheres fixes the shadow issues. Nice!

**Review by marctem — APPROVED** [review]

**Review by oleks-k — APPROVED** [review]
Looks good.

**Pull request merged**